src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -1363,6 +1367,7 @@ public String build(final Path directory, final String name, final String docker
 
     WebTarget resource = noTimeoutResource().path("build");
 
+    final RegistryAuth registryAuth = registryAuthSupplier.authFor(name);
```

Comment:
I don't think this behavior is correct. When building an image, you are supposed to send all the RegistryAuths that the client is configured for: https://docs.docker.com/engine/api/v1.28/#operation/ImageBuild

It might be that the image you are building `FROM` has different RegistryAuth details than the name of the image you are building